src/inmanta/server/services/resourceservice.py

```diff
@@ -335,9 +337,10 @@ async def get_resource_increment_for_agent(self, env: data.Environment, agent: s
 
         now = datetime.datetime.now().astimezone()
 
+        ON_AGENT_REGEX = re.compile(rf"^[a-zA-Z0-9_:]+\[{re.escape(agent)},")
+
         def on_agent(res: ResourceIdStr) -> bool:
-            idr = Id.parse_id(res)
-            return idr.get_agent_name() == agent
+            return ON_AGENT_REGEX.match(res) is not None
```

Comment:
optimization because this is called a on every resource for every agent ( about `o(n**2)` )